### PR TITLE
sys_fs & sys_game: Misc syscalls enhancement

### PR DESCRIPTION
**Pull request by brian218**
1. Fixed potential bugs in sys_game_get_system_sw_version().
2. Implemented sys_game_set_system_sw_version().
3. Improved the efficiency of the conversion from device names to mount points for sys_fs_mount() and sys_fs_newfs().
4. Implemented mount_info_map in sys_fs to keep proper track of mounted devices.
5. Further implemented cellFsArcadeHddSerialNumber (0xc0000007) so that the virtual HDD's model name and serial number are now configurable in config.yml.

### CODE REVIEW

File: rpcs3/Emu/Cell/lv2/sys_game.cpp [hunk 1]
```diff
@@ -211,7 +211,14 @@ error_code _sys_game_watchdog_clear()
 
 u64 _sys_game_get_system_sw_version()
 {
-	return stof(utils::get_firmware_version()) * 10000;
+	try
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/lv2/sys_game.cpp:214]: What are you trying to do here? Since exceptions are disabled, I think it will only result in std::terminate

Comment (Megamouse) [reply rpcs3/Emu/Cell/lv2/sys_game.cpp:214]: I think I implemented fmt::stringtofloat or something like that

Comment (brian218) [reply rpcs3/Emu/Cell/lv2/sys_game.cpp]: I worked around.
The emulator shouldn't be crashed simply due to a bad /dev_flash/vsh/etc/version.txt.

Comment (Megamouse) [reply rpcs3/Emu/Cell/lv2/sys_game.cpp]: Maybe try_to_float in StrUtil.h

File: rpcs3/Emu/Cell/lv2/sys_game.cpp [hunk 2]
```diff
@@ -211,7 +211,9 @@ error_code _sys_game_watchdog_clear()
 
 u64 _sys_game_get_system_sw_version()
 {
-	return stof(utils::get_firmware_version()) * 10000;
+	if (const auto version_str = utils::get_firmware_version(); version_str.find_first_not_of(".0123456789") == umax)
```
[the remark this answers is not in the document]
Comment (brian218) [reply rpcs3/Emu/Cell/lv2/sys_game.cpp]: Done.

**Comment by brian218** [comment]
Image
I observed there are chances that `sys_game_get_system_sw_version()`, which in turn calls `get_firmware_version()`, is called repeatedly, so I added cache to `get_firmware_version()` in order to avoid repetitive IO operation of reading the /dev_flash/vsh/etc/version.txt file.

File: rpcs3/rpcs3qt/main_window.cpp [hunk 3]
```diff
@@ -1554,7 +1554,7 @@ void main_window::HandlePupInstallation(const QString& file_path, const QString&
 	{
 		ui->bootVSHAct->setEnabled(fs::is_file(g_cfg_vfs.get_dev_flash() + "/vsh/module/vsh.self"));
 
-		gui_log.success("Successfully installed PS3 firmware version %s.", version_string);
+		gui_log.success("Successfully installed PS3 firmware version %s.", utils::get_firmware_version(false));
```
Comment (Megamouse) [inline rpcs3/rpcs3qt/main_window.cpp:1557]: don't change this

File: rpcs3/util/sysinfo.cpp [hunk 4]
```diff
@@ -395,59 +395,69 @@ std::string utils::get_system_info()
 	return result;
 }
 
-std::string utils::get_firmware_version()
+std::string utils::get_firmware_version(bool cache)
```
Comment (Megamouse) [inline rpcs3/util/sysinfo.cpp:398]: This makes no sense at all.
Just use fxo to cache it in sys_game

Comment (brian218) [reply rpcs3/util/sysinfo.cpp]: Done 🫡

File: rpcs3/Emu/Cell/lv2/sys_fs.h [hunk 5]
```diff
@@ -161,6 +161,35 @@ struct lv2_fs_mount_point
...
+struct lv2_fs_mount_info
+{
+	lv2_fs_mount_point* const mp;
+	const std::string device;
+	const std::string file_system;
+	const bool read_only;
+
+	lv2_fs_mount_info(lv2_fs_mount_point* mp, std::string_view device = {}, std::string_view file_system = {}, bool read_only = false)
+		: mp(mp)
+		, device(device.empty() ? mp->device : device)
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_fs.h:173]: missing check if mp is null

Comment (brian218) [reply rpcs3/Emu/Cell/lv2/sys_fs.h]: Fixed.

**Comment by jgt11** [comment]
Any testing needed?

**Comment by brian218** [comment]
> Any testing needed?

I don't think so.
Not sure what else is bothering.
It has already been ready for weeks.

**Comment by FlexBy420** [comment]
Seems like this PR broke XMB, it crashes almost after loading SPU Cache
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/11412814/RPCS3.log.gz)

**Comment by brian218** [comment]
> Seems like this PR broke XMB, it crashes almost immediately after loading SPU Cache [RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/11412814/RPCS3.log.gz)

Not a regression.
Kindly clear the caches, and everything should continue to work.

**Pull request merged**